052-cruise-control-api-users.md

```diff
@@ -0,0 +1,269 @@
+
+* `admin`      - Used by Strimzi for Cruise Control admin operations.
+* `heathcheck` - Used by Strimzi for checking the readiness of the Cruise Control application.
+
+Access to these API users is limited to Strimzi.
+So when HTTP basic authentication is enabled for Cruise Control, no one can access the Cruise Control API other than Strimzi.
+
+### How API user management is handled elsewhere
+
+Cruise Control is not the only Kafka component which has API users.
+Although this proposal is stictly about Cruise Control API users, for consistency across the code base, we also considered the following components when designing our solution:
+
+* Kafka users
+* KafkaConnect users
+
+#### Kafka users
+
+In Strimzi, Kafka users are managed via custom resources.
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaUser
+metadata:
+  name: my-user-1
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: tls
+  authorization:
+    type: simple
+    acls:
+      ...
+```
+These resources are watched and managed by the the Strimzi User Operator, which creates Kafka users based on the descriptions provided by the resources.
+
+For more details, checkout the Strimzi documentation [here](https://strimzi.io/docs/operators/latest/deploying#proc-configuring-kafka-user-str)
+
+#### KafkaConnect users
+
+At this time, Strimzi doesn't offer any method of creating or configuring KafkaConnect API users.
+That being said, the direction we take with creating and managing Cruise Control API users could be used as a precedent for its design in the future.
+
+## Motivation
+
+There are a couple use cases where developers would want to access the Cruise Control API without disabling API security:
+
+* Monitoring a Strimzi-managed Kafka cluster with the Cruise Control UI.
+* Gathering Cruise Control specific statistical information that is not currently available via Strimzi Operator or Cruise Control sensor metrics e.g. detailed information surrounding cluster and partition load and user tasks.
+* Experimenting with Cruise Control operations not currently supported by Strimzi.
+* Debugging Cruise Control in a secured environment.
+
+At this time, Strimzi Cruise Control integration offers limited access to Cruise Control data and features.
+This limited access protects developers from preforming cluster operations that coud interfere with Strimzi Operator operations and cause damage to their clusters.
+This limited access also prevents developers from accessing harmless monitoring data and enabling third party applications such as the Cruise Control UI which could make monitoring their Cruise Control data easier.
+
+In order to allow developers and third-party applications to access the Cruise Control API without needing to disable API security, we would need to provide a method of creating API users.
+
+## Proposal
+
+### Goals
+
+* A method of creating Cruise Control API users.
+
+### Developer managed API user configuration
```

Comment:
In this section you're talking a lot about "developers". It's not really what kind of developer you're really thinking of. The motivation section, and the reliance on the `Kafka` CR makes it seem like this more aimed at "Kafka cluster admin", than "Kafka application developers" (i.e. people writing producer and consumer applications)

Reply:
I found "user", "operator", and "admin" to be overloaded terms, terms that conflicted with key parts of the proposal. Yes this proposal is more aimed at Strimzi users, Strimzi operators, Strimzi cluster admins, or we could even say Strimzi devops but not Strimzi or Kafka application developers. However, it seemed that the term "developers" would cause the least confusion here.